### PR TITLE
chore(data-warehouse): data warehouse s3 tables in CTE

### PR DESCRIPTION
**Pull request by EDsCODE**
## Problem

- round 2: joins don't work with s3 tables. When directly joining, the s3tables, incorrect columns will be queried due to clickhouse implementation of s3Cluster. CTEs resolve this
- NOTE: Joining distributed table <- s3Cluster table on a multi-node clickhouse cluster [doesn't work](https://github.com/ClickHouse/ClickHouse/issues/52022)

## Changes

- WIP: put s3 tables into CTEs

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Review by mariusandra — COMMENTED** [review]
How about a test to catch the regression we just experienced?

**Comment by EDsCODE** [comment]
done

**Comment by mariusandra** [comment]
I think there's still one big vulnerability in here. We can't ever use `%` signs inside identifiers, because ClickHouse will complain about missing escape tags, in turn letting people inject whatever (like `%(hogql_val_1)`). I have [some code here](https://github.com/PostHog/posthog/blob/d8fff21942c6cab419ca2899ed9f3aeab7273e03/posthog/hogql/printer.py#L565) blocking it for normal aliases, but I'm not sure if this now goes through that code path.

This probably should be fixed inside `print_identifier` to be really sure nothing slips by.

File: posthog/hogql/printer.py [hunk 1]
```diff
@@ -808,6 +806,9 @@ def _last_select(self) -> Optional[ast.SelectQuery]:
         return None
 
     def _print_identifier(self, name: str) -> str:
```
Comment (EDsCODE) [inline posthog/hogql/printer.py:808]: Moved it to print identifier + added test for select on s3table path

**Review by mariusandra — APPROVED** [review]

**Pull request merged**